public/app/core/components/TimeSeries/utils.ts

```diff
@@ -157,17 +168,31 @@ export const preparePlotConfigBuilder: UPlotConfigPrepFn<{
       builder.addHook('drawAxes', (u: uPlot) => {
         u.ctx.save();
 
-        u.ctx.fillStyle = theme.colors.text.primary;
-        u.ctx.textAlign = 'left';
-        u.ctx.textBaseline = 'bottom';
-
         let i = 0;
         u.axes.forEach((a) => {
-          if (a.side === 2) {
+          if (isVertical && a.side === 2) {
+            u.ctx.fillStyle = theme.colors.text.primary;
+            u.ctx.textAlign = 'left';
+            u.ctx.textBaseline = 'bottom';
             //@ts-ignore
             let cssBaseline: number = a._pos + a._size;
             u.ctx.fillText(timeZones[i], u.bbox.left, cssBaseline * uPlot.pxRatio);
             i++;
+            // In a case we have horizontal orientation, we want to render timezone
+            //  labels on top and want to have them short
+          } else if (!isVertical && a.side === 3) {
+            u.ctx.fillStyle = theme.colors.text.primary;
+            u.ctx.textAlign = 'start';
+            u.ctx.textBaseline = 'top';
+            //@ts-ignore
+            let cssBaseline: number = a._pos - a._size;
+            let timeZone = getZone(timeZones[i])?.abbrs[0] ?? timeZones[i];
+            if (timeZone === 'browser') {
+              const offset = new Date().getTimezoneOffset();
+              timeZone = `UTC${offset < 0 ? '+' : '-'}${Math.abs(offset / 60)}`;
+            }
+            u.ctx.fillText(timeZone, cssBaseline * uPlot.pxRatio, u.bbox.top);
+            i++;
```

Comment:
i'd probably do this in the dom with a new uplot plugin rather than on canvas.

Reply:
Yeah, this can be refactored. For now, I was just following the current implementation. But noted! :)